Source/WebKit/UIProcess/Cocoa/UIDelegate.mm

```diff
@@ -844,6 +844,50 @@ static _WKAutoplayEvent toWKAutoplayEvent(WebCore::AutoplayEvent event)
     return [(id<WKUIDelegatePrivate>)m_uiDelegate->m_delegate.get() _focusWebViewFromServiceWorker:m_uiDelegate->m_webView.get().get()];
 }
 
+bool UIDelegate::UIClient::runOpenPanel(WebPageProxy& page, WebFrameProxy* webFrameProxy, FrameInfoData&& frameInfo, API::OpenPanelParameters* openPanelParameters, WebOpenPanelResultListenerProxy* listener)
+{
+    if (!m_uiDelegate)
+        return false;
+
+    if (!m_uiDelegate->m_delegateMethods.webViewRunOpenPanelWithParametersInitiatedByFrameCompletionHandler)
+        return false;
+
+    auto delegate = m_uiDelegate->m_delegate.get();
+    if (!delegate)
+        return false;
+
+    auto frame = API::FrameInfo::create(WTFMove(frameInfo), &page);
+
+    auto checker = CompletionHandlerCallChecker::create(delegate.get(), @selector(webView:runOpenPanelWithParameters:initiatedByFrame:completionHandler:));
```

Comment:
If these return Ref or RefPtrs, they should use that instead of auto

Reply:
This method was actually just relocated to avoid the platform check, I didn't write it or make any changes. Is it worth updating still? @rr-codes

Reply:
Those were using auto before my changes; is it worth updating anyways? @rr-codes

Reply:
Yeah, `auto` -> explicit smart pointer is a nice change to make when moving code.